src/compiler/checker.ts

```diff
@@ -8910,7 +8911,11 @@ namespace ts {
         }
 
         function isTupleLikeType(type: Type): boolean {
-            return !!getPropertyOfType(type, "0");
+            return !!getPropertyOfType(type, "0") || isEmptyTupleType(type);
+        }
+
+        function isEmptyTupleType(type: Type): boolean {

```

Comment:
you should be to use `===` directly, so I think you should drop `isEmptyTupleType` and just add `|| type === emptyTupleType` on line 8914